index.js

```diff
@@ -36,54 +37,125 @@ function elbMetrics(startTime, endTime, region, elbname, callback) {
         return callback(new Error('start and end time should not be more than 60 minutes apart'));
     }
 
-    var parameters = {
-        startTime: startTime,
-        endTime: endTime,
-        region: region,
-        elbname: elbname
-    };
-
-    var queries = prepareQueries(parameters);
-    outputMetrics(queries, region, function (err, data) {
+    preFlightCheck(elbname, function (err, name) {
+        var elb = 0;
         if (err) return callback(err);
-        var requestPercentages = prepareResults(startTime, endTime, data);
-        callback(null, requestPercentages);
+        if (name.indexOf('app/') === -1) { elb = 1; }
+        var parameters = {
+            startTime: startTime,
+            endTime: endTime,
+            region: region,
+            elbname: name
+        };
+        var queries = prepareQueries(parameters, elb);
+        outputMetrics(queries, region, function (err, data) {
+            if (err) return callback(err);
+            var requestPercentages = prepareResults(startTime, endTime, data);
+            return callback(null, requestPercentages);
+        });
+    });
+}
+
+function preFlightCheck(elbname, callback) {
+    var q = queue(6);
+    q.defer(getELBName, elbname);
+    q.defer(getALBName, elbname);
+    q.awaitAll(function (err, data) {
+        if (err) console.log('err', err);
+        data = data.filter(d => { return d; });
+        return callback(null, data[0]);
+
+    });
+}
+
+function getALBName(elbname, callback) {
+    var elbv2 = new AWS.ELBv2();
+    var parameter = {Names: [elbname]};
+    elbv2.describeLoadBalancers(parameter, function (err, data) {
+        if (err) {
+            if (err.code === 'LoadBalancerNotFound') return callback(null, null);
```

Comment:
[style] `return callback(null, null)` can also be expressed as `return callback()`